pom.xml

```diff
@@ -364,7 +364,7 @@
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-surefire-plugin</artifactId>
         <configuration>
-          <parallel>classes</parallel>
+          <parallel>suites</parallel>
```

Comment:
was this intentionally? I guess this implies that we don't try run tests in parallel or do we utilize suites?

Reply:
Yes, that's intentionally. Because of tests not passing, caused by SingletonRandomGenerator, which can not run in parallel...
Maybe  <parallel>suites</parallel> is not the best solution. Rather, removing parallel altogether?

Reply:
I tried to get around the problem, using [ThreadLocal](https://stackoverflow.com/questions/14959295/threadlocal-singleton), and also dependency injection. But I did not find any really good solution to the parallel tests issues

Reply:
I think it would be better to clearly remove parallel execution, otherwise it would be easy to trick ourself looking through the configuration. Or if we add suites it might start to fail. How big is the difference in build time, do we need parallel?

Reply:
I [removed parallel execution](https://github.com/GraphWalker/graphwalker-project/pull/218/commits/677a86579caebb042756c8e4f6cfc981c04774d2) altogether, and compared with execution on master. After warming up, I compared 2 runs with `time mvn test`, and got:
```
Master with parallel test execution
1 - 6m34,882s
2 - 6m32,750s

This PR without parallel test execution
1 - 6m33,411s
2 - 6m20,338s
```
So, oddly enough, there was no performance degradation. In fact,one run executed faster.